source/common/upstream/thread_aware_lb_impl.h

```diff
@@ -28,6 +29,43 @@ class ThreadAwareLoadBalancerBase : public LoadBalancerBase, public ThreadAwareL
   };
   using HashingLoadBalancerSharedPtr = std::shared_ptr<HashingLoadBalancer>;
 
+  class BoundedLoadHashingLoadBalancer : public HashingLoadBalancer {
+  public:
+    /**
+     * Class for consistent hashing load balancer (CH-LB) with bounded loads.
+     * It is common to both RingHash and Maglev load balancers, because the logic of selecting the
+     * next host when one is overloaded is independent of the CH-LB type.
+     */
+    BoundedLoadHashingLoadBalancer(HashingLoadBalancerSharedPtr hashing_lb_ptr,
+                                   const NormalizedHostWeightVector& normalized_host_weights,
```

Comment:
if you're gonna move this, pass it by value or rvalue ref (&&). Right now you're making a copy

Reply:
Can you please check now? I am now passing _rvalue_ in for ```normalized_host_weights``` every call so that it is always moved. Would this correct?

From ```ThreadAwareLoadBalancerBase::refresh()``` to ```RingHashLoadBalancer::createLoadBalancer``` to ```BoundedLoadHashingLoadBalancer::ctor```